Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1175,9 +1136,26 @@ class CombinatorialKalmanFilter {
     auto propState =
         m_propagator.template makeState(initialParameters, propOptions);
 
-    auto& r = propState.template get<CombinatorialKalmanFilterResult<traj_t>>();
-    r.fittedStates = &trackContainer.trackStateContainer();
-    r.stateBuffer = std::make_shared<traj_t>();
+    auto& r =
+        propState
+            .template get<CombinatorialKalmanFilterResult<track_container_t>>();
+    r.tracks = &trackContainer;
+    r.trackStates = &trackContainer.trackStateContainer();
```

Comment:
This are a bit of a code smell, keeping pointers to a `findTracks` argument in the result. It was there before this PR, but now we have an extra pointer.
Would it be possible to remove this and pass `trackContainer` through to where it is needed? I guess not, but just thought I'd check. Alternatively, would we lose much by removing the `trackStates` pointer?

Reply:
Yes this is not great. The lifetime should be ensured since these pointers are only used while we are inside the `findTracks` function.

Reply:
I guess you can get the track state container from the track container? Do we need the extra pointer?